tools/pipelines/templates/build-npm-package.yml

```diff
@@ -264,16 +264,11 @@ stages:
                 fi
               fi
 
-              if [[ "$(release)" != "none" ]] && [[ "$(release)" != "" ]]; then
+              if [[ "$(release)" != "none" ]] || [[ "$(release)" != "" ]]; then
                 if [[ "${{ variables.publish }}" != "True" ]]; then
                   echo "##vso[task.logissue type=error]'$(release)'' is set but package is not published. Either the branch doesn't default to publish or it is skipped."
```

Comment:
Is the intention here to immediately print the value of `$(release)`? If not, we probably need some escaping so ADO doesn't just replace the variable with its value.

Reply:
I tried to reframe this error message. If release is none or empty and publish is set to true, then error out.